### PR TITLE
Remove "laravel-" prefix from config & merge config

### PR DESCRIPTION
**Pull request by m1guelpf**
Yes, it's a breaking change, but it's a good practice and we have semver for that.

### CODE REVIEW

**Comment by m1guelpf** [comment]
@joshcanhelp Done. The old config should still work, but the new one will be used for new installs.

**Comment by joshcanhelp** [comment]
Thanks again @m1guelpf! 

And, just to make sure, this won't fail if one of those paths doesn't exist?

**Comment by m1guelpf** [comment]
nope :)

**Comment by joshcanhelp** [comment]
Sounds good. I'll run this through a quick test today/tomorrow and merge this in!

**Comment by joshcanhelp** [comment]
@m1guelpf - So ... I don't think this is working as expected. You added:

```php
// 
class LoginServiceProvider extends ServiceProvider {
    // ...
    public function register()
    {
        // ...
        $this->mergeConfigFrom(__DIR__.'/../../config/config.php', 'auth0');
        $this->mergeConfigFrom(__DIR__.'/../../config/config.php', 'auth0-laravel');
    }
    // ...
}
```

... and from what I understand about how `mergeConfigFrom` work ([ref](https://laracasts.com/discuss/channels/general-discussion/how-does-mergeconfigfrom-work)), the first parameter is the file you want to merge from and the second is the config key. The config key for this package is `laravel-auth0` so this merge doesn't do anything. 

The other problem is that you're merging _from_ the default config so if I change the config key to the proper one mentioned above, I'm now merging in the default config file that comes with the package, not the one we want to support, `config/laravel-auth0.php`. I can confirm that's the case by renaming `laravel-auth0.php` to `auth0.php`. It picks up the default callback URL (which is different in the test app I'm running). 

If I adjust everything to point to the right files, I see that this does not check for the existence of the file before attempting to merge the settings in. So we need to check for the the `auth0.php` file first:

```php
// 
class LoginServiceProvider extends ServiceProvider {
    // ...
    public function register()
    {
        // ...
        if ( file_exists( app_path() . '/../config/auth0.php' ) ) {
            $this->mergeConfigFrom(app_path() . '/../config/auth0.php', 'laravel-auth0');
        }
    }
    // ...
}
```

If I do that, it works with the file at either location. 

Thoughts?

**Comment by joshcanhelp** [comment]
Note for myself: we'll need to update this section in the Quickstart when/if the file changes:

https://auth0.com/docs/quickstart/webapp/laravel/01-login#configure-it

**Comment by m1guelpf** [comment]
@joshcanhelp Everything should work now

**Comment by joshcanhelp** [comment]
@m1guelpf - But, again, that's merging in the defaults from the package, not the user-maintained file with a different name. Am I missing something here?

**Comment by m1guelpf** [comment]
@joshcanhelp updated fallback

File: src/Auth0/Login/LoginServiceProvider.php [hunk 1]
```diff
@@ -64,6 +64,12 @@ public function register()
         \Event::listen('Illuminate\Auth\Events\Logout', function () {
             \App::make('auth0')->logout();
         });
+        
+        $this->mergeConfigFrom(config_path('auth0.php'), 'auth0');
```
Comment (joshcanhelp) [inline src/Auth0/Login/LoginServiceProvider.php]: As mentioned before, this will fail if the file does not exist.

Comment (joshcanhelp) [reply src/Auth0/Login/LoginServiceProvider.php]: Also, `auth0` is not the settings name that's used throughout the package (cannot change without a BC)

File: src/Auth0/Login/LoginServiceProvider.php [hunk 2]
```diff
@@ -64,6 +64,12 @@ public function register()
...
+        
+        if (file_exists(config_path('auth0-laravel.php')) {
+            $this->mergeConfigFrom(config_path('auth0-laravel.php'), 'auth0');
```
Comment (joshcanhelp) [inline src/Auth0/Login/LoginServiceProvider.php]: I don't know where this file name is coming from, was previously `auth0-laravel.php`. Also: setting name.

File: src/Auth0/Login/LoginServiceProvider.php (hunk 1 above)
Comment (m1guelpf) [reply src/Auth0/Login/LoginServiceProvider.php]: That command doesn't need the file to exist afaiq, and there's a fallback for the old config key

File: src/Auth0/Login/LoginServiceProvider.php (hunk 2 above)
Comment (m1guelpf) [reply src/Auth0/Login/LoginServiceProvider.php]: So here I'm merging the old config into the new key for backwards compat

File: src/Auth0/Login/LoginServiceProvider.php (hunk 1 above)
Comment (joshcanhelp) [reply src/Auth0/Login/LoginServiceProvider.php]: > there's a fallback for the old config key

Where is that? Right now, the setting key is called `laravel-auth0`.

File: src/Auth0/Login/LoginServiceProvider.php (hunk 2 above)
Comment (joshcanhelp) [reply src/Auth0/Login/LoginServiceProvider.php]: `auth0-laravel.php` is not the old settings file name ;)

File: src/Auth0/Login/LoginServiceProvider.php (hunk 1 above)
Comment (m1guelpf) [reply src/Auth0/Login/LoginServiceProvider.php]: Fixed

Comment (joshcanhelp) [reply src/Auth0/Login/LoginServiceProvider.php]: @m1guelpf - What happens when someone already using the package calls, for example, `config('laravel-auth0.client_id')`? Will that work after this update?

Comment (m1guelpf) [reply src/Auth0/Login/LoginServiceProvider.php]: Fixed

**Comment by joshcanhelp** [comment]
@m1guelpf - Thanks for sticking with it. Just to make sure ... did you test this manually with both keys and both file locations? We need to makes sure this doesn't break anything when updated. Let me know and I'll take a final walk-through before merging.

**Comment by joshcanhelp** [comment]
Closing, no response.

**Pull request closed without merging**